server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -351,12 +358,25 @@ private List<FlowInstanceDetailResp> innerCreate(@NotNull @Valid CreateFlowInsta
         List<RiskLevel> riskLevels = riskLevelService.list();
         Verify.notEmpty(riskLevels, "riskLevels");
         ConnectionConfig conn = null;
-        if (Objects.nonNull(createReq.getConnectionId())) {
+        List<ConnectionConfig> conns = null;
+        if (Objects.nonNull(createReq.getConnectionId()) && createReq.getTaskType() != TaskType.MULTIPLE_ASYNC) {
             conn = connectionService.getForConnectionSkipPermissionCheck(createReq.getConnectionId());
             cloudMetadataClient.checkPermission(OBTenant.of(conn.getClusterName(),
                     conn.getTenantName()), conn.getInstanceType(), false, CloudPermissionAction.READONLY);
+        } else {
+            MultipleDatabaseChangeParameters taskParameters =
+                    (MultipleDatabaseChangeParameters) createReq.getParameters();
+            conns = taskParameters.getDatabases().stream().map(
```

Comment:
getForConnectionSkipPermissionCheck may involve performance problem.

may group by datasourceId, due multiple databases may share same datasource config.

Reply:
All non-duplicate data source ids have been selected, and then the data source is verified